src/test/java/com/epam/training/student_gregory_kubya/MainPage.java

```diff
@@ -1,13 +1,57 @@
 package com.epam.training.student_gregory_kubya;
+
+import java.time.Duration;
+import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
 
-// page_url =
 public class MainPage {
 
-    public MainPage(WebDriver driver) {
-        PageFactory.initElements(driver, this);
-    }
+  private final WebDriver driver;
+  private static final String pageURL = "https://fiftyoutlet.com/";
+
+  @FindBy(id = "onetrust-accept-btn-handler")
+  public WebElement acceptCookieButton;
+
+  @FindBy(css = "div.navigation.navigation-search")
+  public WebElement searchButton;
+
+  @FindBy(css = ".gi-search-search-input")
+  public WebElement searchField;
+
+  @FindBy(css = "div[data-dl-search-count]")
+  public WebElement searchCount;
+
+  public void searchForValue(String value) {
+    this.searchButton.click();
+    this.searchField.sendKeys(value);
```

Comment:
I suggest adding a waiter to ensure searchField is displayed before sending keys.